src/commands/ext-dev-unpublish.ts

```diff
@@ -0,0 +1,47 @@
+import { Command } from "../command";
+import { logPrefix } from "../extensions/extensionsHelper";
+import { unpublishExtension, parseRef, getExtension } from "../extensions/extensionsApi";
+import * as utils from "../utils";
+import { promptOnce } from "../prompt";
+import * as clc from "cli-color";
+import { requireAuth } from "../requireAuth";
+
+module.exports = new Command("ext:dev:unpublish [ref]")
```

Comment:
This `ref` doesn't match the `publisherId` and `extensionId` pair of parameters as set out in `:publish`. Is there a reason for this?

Reply:
Good point. This was initially supposed to support two ref formats:
1) extension ref: publisherId/extensionId
2) version ref: publisherId/extensionId@versionId

However we de-scoped un-publishing a specific version because it introduced several edge cases. That's why the code looks like this.

But you're right - I refactored code in `ext:dev:publish` to use parseRef, and pass in extensionRef instead of two values.